aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -180,23 +181,94 @@ struct CacheKeyFusedWrapper : ParamsWrapper<CacheKeyFused> {
   }
 };
 
+static int _parseChosenLRUCacheLimit() {
+  const char * val = getenv("TORCH_CUDNN_V8_API_LRU_CACHE_LIMIT");
+  int limit = 10000; // roughly corresponds to 2GiB assuming 200KiB per ExecutionPlan
+  if (val) {
+    try {
+      limit = std::stoi(val);
+    } catch(std::invalid_argument const& e) {
+      TORCH_WARN("invalid TORCH_CUDNN_V8_API_LRU_CACHE_LIMIT,",
+                 " using default LRU cache limit of ", limit, " entries.");
+    } catch(std::out_of_range const& e) {
+      TORCH_WARN("invalid TORCH_CUDNN_V8_API_LRU_CACHE_LIMIT,",
+                 " using default LRU cache limit of ", limit, " entries.");
+    }
+  }
+  return limit;
+}
+
+static int getLRUCacheLimit() {
+  static int limit = _parseChosenLRUCacheLimit();
+  return limit;
+}
+
 template <typename T, typename KeyType>
 struct BenchmarkCache {
-std::unordered_map<KeyType, cudnn_frontend::ExecutionPlan, ParamsWrapperHash<KeyType>> engine_cache;
+std::list<KeyType> engine_cache_order;
+std::unordered_map<KeyType, std::pair<cudnn_frontend::ExecutionPlan, typename std::list<KeyType>::iterator>, ParamsWrapperHash<KeyType>> engine_cache;
 
 // no mutexes here as caches are now thread local for v8, can also return a pointer
 // to the Execution Plan if we know it will not be invalidated by another thread
 cudnn_frontend::ExecutionPlan* find(const KeyType& key) {
+  int lru_cache_limit = getLRUCacheLimit();
+  if (lru_cache_limit < 0) {
+    return nullptr;
+  }
   auto it = engine_cache.find(key);
   if (it == engine_cache.end()) {
     return nullptr;
   }
-  return &(it->second);
+  if (lru_cache_limit) {
+    TORCH_INTERNAL_ASSERT(*(it->second.second) == key, "CUDNN V8 LRU Cache Corrupted (found key mismatches list). Please report a bug to PyTorch.");
+    auto engine_cache_order_size = engine_cache_order.size();
+    auto engine_cache_size = engine_cache.size();
+    TORCH_INTERNAL_ASSERT(engine_cache_order_size == engine_cache_size, "CUDNN V8 LRU Cache Corrupted (found list vs. map size mismatch). Please report a bug to PyTorch.");
+    // update most recently accessed
+    auto plan = it->second.first;
+    engine_cache_order.erase(it->second.second);
```

Comment:
This is O(N), isn't it? Is there a more efficient method to implement LRU? Feels like [`std::list::splice`](https://en.cppreference.com/w/cpp/container/list/splice) and unordered map of keys to iterators in the list (which are not invalidated as result of splicing) feels like a better solution, isn't it?

Reply:
Add/removing nodes in the list should not be O(N) as `std::List` is backed by a doubly-linked-list rather than a contiguous array, unless you are referring to something else. Note that we are selecting the node to remove with an iterator as well which should not require any traversal.